File: android/src/main/java/com/spoonacular/client/model/RecipeInformationWinePairingProductMatchesInner.java
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.1
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

package com.spoonacular.client.model;

import java.math.BigDecimal;
import io.swagger.annotations.*;
import com.google.gson.annotations.SerializedName;

@ApiModel(description = "")
public class RecipeInformationWinePairingProductMatchesInner {
  
  @SerializedName("id")
  private Integer id = null;
  @SerializedName("title")
  private String title = null;
  @SerializedName("description")
  private String description = null;
  @SerializedName("price")
  private String price = null;
  @SerializedName("imageUrl")
  private String imageUrl = null;
  @SerializedName("averageRating")
  private BigDecimal averageRating = null;
  @SerializedName("ratingCount")
  private Integer ratingCount = null;
  @SerializedName("score")
  private BigDecimal score = null;
  @SerializedName("link")
  private String link = null;

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Integer getId() {
    return id;
  }
  public void setId(Integer id) {
    this.id = id;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getTitle() {
    return title;
  }
  public void setTitle(String title) {
    this.title = title;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getDescription() {
    return description;
  }
  public void setDescription(String description) {
    this.description = description;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getPrice() {
    return price;
  }
  public void setPrice(String price) {
    this.price = price;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getImageUrl() {
    return imageUrl;
  }
  public void setImageUrl(String imageUrl) {
    this.imageUrl = imageUrl;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getAverageRating() {
    return averageRating;
  }
  public void setAverageRating(BigDecimal averageRating) {
    this.averageRating = averageRating;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Integer getRatingCount() {
    return ratingCount;
  }
  public void setRatingCount(Integer ratingCount) {
    this.ratingCount = ratingCount;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getScore() {
    return score;
  }
  public void setScore(BigDecimal score) {
    this.score = score;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getLink() {
    return link;
  }
  public void setLink(String link) {
    this.link = link;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    RecipeInformationWinePairingProductMatchesInner recipeInformationWinePairingProductMatchesInner = (RecipeInformationWinePairingProductMatchesInner) o;
    return (this.id == null ? recipeInformationWinePairingProductMatchesInner.id == null : this.id.equals(recipeInformationWinePairingProductMatchesInner.id)) &&
        (this.title == null ? recipeInformationWinePairingProductMatchesInner.title == null : this.title.equals(recipeInformationWinePairingProductMatchesInner.title)) &&
        (this.description == null ? recipeInformationWinePairingProductMatchesInner.description == null : this.description.equals(recipeInformationWinePairingProductMatchesInner.description)) &&
        (this.price == null ? recipeInformationWinePairingProductMatchesInner.price == null : this.price.equals(recipeInformationWinePairingProductMatchesInner.price)) &&
        (this.imageUrl == null ? recipeInformationWinePairingProductMatchesInner.imageUrl == null : this.imageUrl.equals(recipeInformationWinePairingProductMatchesInner.imageUrl)) &&
        (this.averageRating == null ? recipeInformationWinePairingProductMatchesInner.averageRating == null : this.averageRating.equals(recipeInformationWinePairingProductMatchesInner.averageRating)) &&
        (this.ratingCount == null ? recipeInformationWinePairingProductMatchesInner.ratingCount == null : this.ratingCount.equals(recipeInformationWinePairingProductMatchesInner.ratingCount)) &&
        (this.score == null ? recipeInformationWinePairingProductMatchesInner.score == null : this.score.equals(recipeInformationWinePairingProductMatchesInner.score)) &&
        (this.link == null ? recipeInformationWinePairingProductMatchesInner.link == null : this.link.equals(recipeInformationWinePairingProductMatchesInner.link));
  }

  @Override
  public int hashCode() {
    int result = 17;
    result = 31 * result + (this.id == null ? 0: this.id.hashCode());
    result = 31 * result + (this.title == null ? 0: this.title.hashCode());
    result = 31 * result + (this.description == null ? 0: this.description.hashCode());
    result = 31 * result + (this.price == null ? 0: this.price.hashCode());
    result = 31 * result + (this.imageUrl == null ? 0: this.imageUrl.hashCode());
    result = 31 * result + (this.averageRating == null ? 0: this.averageRating.hashCode());
    result = 31 * result + (this.ratingCount == null ? 0: this.ratingCount.hashCode());
    result = 31 * result + (this.score == null ? 0: this.score.hashCode());
    result = 31 * result + (this.link == null ? 0: this.link.hashCode());
    return result;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class RecipeInformationWinePairingProductMatchesInner {\n");
    
    sb.append("  id: ").append(id).append("\n");
    sb.append("  title: ").append(title).append("\n");
    sb.append("  description: ").append(description).append("\n");
    sb.append("  price: ").append(price).append("\n");
    sb.append("  imageUrl: ").append(imageUrl).append("\n");
    sb.append("  averageRating: ").append(averageRating).append("\n");
    sb.append("  ratingCount: ").append(ratingCount).append("\n");
    sb.append("  score: ").append(score).append("\n");
    sb.append("  link: ").append(link).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

File: csharp/src/spoonacular/Model/RecipeInformationWinePairingProductMatchesInner.cs
/*
 * spoonacular API
 *
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.1
 * Contact: mail@spoonacular.com
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = spoonacular.Client.OpenAPIDateConverter;

namespace spoonacular.Model
{
    /// <summary>
    /// RecipeInformationWinePairingProductMatchesInner
    /// </summary>
    [DataContract(Name = "RecipeInformation_winePairing_productMatches_inner")]
    public partial class RecipeInformationWinePairingProductMatchesInner : IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="RecipeInformationWinePairingProductMatchesInner" /> class.
        /// </summary>
        [JsonConstructorAttribute]
        protected RecipeInformationWinePairingProductMatchesInner() { }
        /// <summary>
        /// Initializes a new instance of the <see cref="RecipeInformationWinePairingProductMatchesInner" /> class.
        /// </summary>
        /// <param name="id">id (required).</param>
        /// <param name="title">title (required).</param>
        /// <param name="description">description (required).</param>
        /// <param name="price">price (required).</param>
        /// <param name="imageUrl">imageUrl (required).</param>
        /// <param name="averageRating">averageRating (required).</param>
        /// <param name="ratingCount">ratingCount (required).</param>
        /// <param name="score">score (required).</param>
        /// <param name="link">link (required).</param>
        public RecipeInformationWinePairingProductMatchesInner(int id = default(int), string title = default(string), string description = default(string), string price = default(string), string imageUrl = default(string), decimal averageRating = default(decimal), int ratingCount = default(int), decimal score = default(decimal), string link = default(string))
        {
            this.Id = id;
            // to ensure "title" is required (not null)
            if (title == null)
            {
                throw new ArgumentNullException("title is a required property for RecipeInformationWinePairingProductMatchesInner and cannot be null");
            }
            this.Title = title;
            // to ensure "description" is required (not null)
            if (description == null)
            {
                throw new ArgumentNullException("description is a required property for RecipeInformationWinePairingProductMatchesInner and cannot be null");
            }
            this.Description = description;
            // to ensure "price" is required (not null)
            if (price == null)
            {
                throw new ArgumentNullException("price is a required property for RecipeInformationWinePairingProductMatchesInner and cannot be null");
            }
            this.Price = price;
            // to ensure "imageUrl" is required (not null)
            if (imageUrl == null)
            {
                throw new ArgumentNullException("imageUrl is a required property for RecipeInformationWinePairingProductMatchesInner and cannot be null");
            }
            this.ImageUrl = imageUrl;
            this.AverageRating = averageRating;
            this.RatingCount = ratingCount;
            this.Score = score;
            // to ensure "link" is required (not null)
            if (link == null)
            {
                throw new ArgumentNullException("link is a required property for RecipeInformationWinePairingProductMatchesInner and cannot be null");
            }
            this.Link = link;
        }

        /// <summary>
        /// Gets or Sets Id
        /// </summary>
        [DataMember(Name = "id", IsRequired = true, EmitDefaultValue = true)]
        public int Id { get; set; }

        /// <summary>
        /// Gets or Sets Title
        /// </summary>
        [DataMember(Name = "title", IsRequired = true, EmitDefaultValue = true)]
        public string Title { get; set; }

        /// <summary>
        /// Gets or Sets Description
        /// </summary>
        [DataMember(Name = "description", IsRequired = true, EmitDefaultValue = true)]
        public string Description { get; set; }

        /// <summary>
        /// Gets or Sets Price
        /// </summary>
        [DataMember(Name = "price", IsRequired = true, EmitDefaultValue = true)]
        public string Price { get; set; }

        /// <summary>
        /// Gets or Sets ImageUrl
        /// </summary>
        [DataMember(Name = "imageUrl", IsRequired = true, EmitDefaultValue = true)]
        public string ImageUrl { get; set; }

        /// <summary>
        /// Gets or Sets AverageRating
        /// </summary>
        [DataMember(Name = "averageRating", IsRequired = true, EmitDefaultValue = true)]
        public decimal AverageRating { get; set; }

        /// <summary>
        /// Gets or Sets RatingCount
        /// </summary>
        [DataMember(Name = "ratingCount", IsRequired = true, EmitDefaultValue = true)]
        public int RatingCount { get; set; }

        /// <summary>
        /// Gets or Sets Score
        /// </summary>
        [DataMember(Name = "score", IsRequired = true, EmitDefaultValue = true)]
        public decimal Score { get; set; }

        /// <summary>
        /// Gets or Sets Link
        /// </summary>
        [DataMember(Name = "link", IsRequired = true, EmitDefaultValue = true)]
        public string Link { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class RecipeInformationWinePairingProductMatchesInner {\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  Title: ").Append(Title).Append("\n");
            sb.Append("  Description: ").Append(Description).Append("\n");
            sb.Append("  Price: ").Append(Price).Append("\n");
            sb.Append("  ImageUrl: ").Append(ImageUrl).Append("\n");
            sb.Append("  AverageRating: ").Append(AverageRating).Append("\n");
            sb.Append("  RatingCount: ").Append(RatingCount).Append("\n");
            sb.Append("  Score: ").Append(Score).Append("\n");
            sb.Append("  Link: ").Append(Link).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            // Title (string) minLength
            if (this.Title != null && this.Title.Length < 1)
            {
                yield return new ValidationResult("Invalid value for Title, length must be greater than 1.", new [] { "Title" });
            }

            // Description (string) minLength
            if (this.Description != null && this.Description.Length < 1)
            {
                yield return new ValidationResult("Invalid value for Description, length must be greater than 1.", new [] { "Description" });
            }

            // Price (string) minLength
            if (this.Price != null && this.Price.Length < 1)
            {
                yield return new ValidationResult("Invalid value for Price, length must be greater than 1.", new [] { "Price" });
            }

            // ImageUrl (string) minLength
            if (this.ImageUrl != null && this.ImageUrl.Length < 1)
            {
                yield return new ValidationResult("Invalid value for ImageUrl, length must be greater than 1.", new [] { "ImageUrl" });
            }

            // Link (string) minLength
            if (this.Link != null && this.Link.Length < 1)
            {
                yield return new ValidationResult("Invalid value for Link, length must be greater than 1.", new [] { "Link" });
            }

            yield break;
        }
    }

}
